mlflow/types/type_hints.py

```diff
@@ -290,7 +305,9 @@ def _infer_schema_from_list_type_hint(type_hint: type[list[Any]]) -> Schema:
     """
     if not _is_list_type_hint(type_hint):
         raise InvalidTypeHintException(
-            message=f"Type hint for model input must be `list[...]`, got {type_hint}",
+            message="Type hint for model input must be `list[...]`, because "
+            "MLflow pyfunc model's `predict` function is expected to work for "
+            "batch data. Fix this by wrapping your type hint with list[...].",
```

Comment:
"Type hints must be wrapped in `list[...]` to support multiple-record batch inference. Specify your type hint as `list[{_type_hint_repr(type_hint)}` for a valid signature."

Perhaps providing the direct example might simplify things for users (or at least make it directionally less confusing for some?)